src/components/SignupPage.jsx

```diff
@@ -0,0 +1,289 @@
+import { Link, useNavigate } from "react-router-dom";
+import signupImg from "../assets/signup-illu.png";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import * as Yup from "yup";
+import { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { signUp, login } from "../redux/user/userActions";
+import CryptoJS from "crypto-js";
+import { toast, Slide } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+const initialValues = {
+  name: "",
+  email: "",
+  phoneNo: "",
+  password: "",
+  confirmPassword: "",
+  profile: "",
+};
+
+// Define validation schema using Yup
+const validationSchema = Yup.object().shape({
+  name: Yup.string()
+    .required("Name is required")
+    .min(15, "Atleast 15 characters are required!"),
+  email: Yup.string()
+    .required("Email is required")
+    .email("Invalid email address"),
+  phoneNo: Yup.string()
+    .required("Phone number is required")
+    .matches(/^\91[1-9]\d{9}$/, "Invalid phone number"),
+  password: Yup.string()
+    .required("Password is required")
+    .min(8, "Password must be at least 8 characters long"),
+  confirmPassword: Yup.string()
+    .required("Please confirm your password")
+    .oneOf([Yup.ref("password"), ""], "Passwords do not match"),
+  profile: Yup.mixed()
+    .required("Profile picture is required")
+    .test("fileSize", "Too Big! Image only up to 2 MB is allowed", (value) => {
+      if (value && value.size) {
+        const fileSizeInMB = value.size / 1024 / 1024;
+        return fileSizeInMB <= 2;
+      }
+      return true;
+    })
+    .test(
+      "fileType",
+      "Invalid File Format! Only png,jpeg/jpg format is allowed",
+      (value) => {
+        if (value && value.type) {
+          return ["image/png", "image/jpeg", "image/jpg"].includes(value.type);
+        }
+        return true;
+      }
+    ),
+});
+
+const encryption = (password) => {
+  const encrypted = CryptoJS.AES.encrypt(
+    password,
+    import.meta.env.VITE_SECRET_KEY
+  ).toString();
+
+  return encrypted;
+};
+
+const SignupPage = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [imagePreview, setImagePreview] = useState(null);
+
+  const handleImageChange = (event) => {
+    const file = event.target.files[0];
+    if (file) {
+      setImagePreview(URL.createObjectURL(file));
+    } else {
+      setImagePreview(null);
+    }
+  };
+
+  const users = useSelector((state) => state.user.users);
```

Comment:
It would be best if you group all the state variables at the top of the component for better maintainability.